sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3511,6 +3511,17 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
     const AdapterPtr &Adapter = MQueue->getAdapter();
     if (MEvent != nullptr)
       MEvent->setHostEnqueueTime();
+    // Honor explicit dependencies set with depends_on.
+    // Important for cross queue dependencies.
+    // Intentionally do not sort out events on the same queue (no sense).
+    // Sync of events wait to be done by the following barrier
```

Comment:
Can you clarify this please?

Reply:
https://github.com/intel/llvm/pull/18139/commits/c64a71a6df9418858f95d604982728ed6e301136